pkg/provider/garden/ruleset/securityhardenedshoot/rules/1002.go

```diff
@@ -0,0 +1,172 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package rules
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/diki/pkg/rule"
+	"github.com/gardener/diki/pkg/shared/ruleset/disak8sstig/option"
+)
+
+var (
+	_ rule.Rule     = &Rule1002{}
+	_ rule.Severity = &Rule1002{}
+	_ option.Option = &Options1002{}
+)
+
+type Rule1002 struct {
+	Client         client.Client
+	ShootName      string
+	ShootNamespace string
+	Options        *Options1002
+}
+
+type Options1002 struct {
+	MachineImages []MachineImage `json:"machineImages" yaml:"machineImages"`
+}
+
+type MachineImage struct {
+	Name                    string                                    `json:"name" yaml:"name"`
+	ExpectedClassifications []gardencorev1beta1.VersionClassification `json:"expectedClassifications" yaml:"expectedClassifications"`
```

Comment:
WDYT of renaming this to allowed? We already use the term 'allowed' in several other places.

```suggestion
	AllowedClassifications []gardencorev1beta1.VersionClassification `json:"allowedClassifications" yaml:"allowedClassifications"`
```